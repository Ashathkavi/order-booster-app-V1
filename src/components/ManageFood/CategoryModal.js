import React from 'react'
import Modal from 'react-modal'
import CategoryList from '../../fixtures/FoodCategoryList'

const CategoryModal = (props) => {
    return(
        <Modal
            isOpen={props.modalIsOpen}
            contentLabel = "Please Select a Category"
            onRequestClose = {props.handleCloseModal}
        >
            <h3>Please Select a Category</h3>
            {
                CategoryList().map((category)=>(
                    <button onClick={() =>{                
                        props.handleCategory(category)
                        props.handleCloseModal()
                    }}>{category}</button>
                ))
            }
            
        </Modal>
)}

export default CategoryModal;


/*

*/