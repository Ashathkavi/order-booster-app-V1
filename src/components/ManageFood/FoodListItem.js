import React, {useState} from 'react'
import { connect } from 'react-redux'
import {removeFood} from '../../actions/foods'
import {Link} from 'react-router-dom'

class SingleTypeOrder {
    constructor(food, foodQuantity){
        this.food= food
        this.foodQuantity= foodQuantity
    }
}

export const FoodListItem = ({
    id, dispatch, name, category, largeAvailability, 
    foodSize, description, amount, fromOrderModal, onAddsingleTypeOrder
}) => {
    const [foodQuantity, setFoodQuantity] = useState(0)

    if(fromOrderModal){
        //console.log('foodQuantity', foodQuantity)
        

        const food = {name, id, category, amount, foodSize}
        const singleTypeOrder = new SingleTypeOrder(food,foodQuantity )
        //console.log('singleTypeOrder', singleTypeOrder)
        return(
            <div>
                <h3>{name}::::: {foodSize}::::: {amount}:::::::{category}</h3>
                <input type='number' max='20' value={foodQuantity} onChange={(e)=>setFoodQuantity(e.target.value)}/>
                <button type="button" onClick={()=>onAddsingleTypeOrder(singleTypeOrder)}>Select</button>
            </div>
        )
    }

    return(
        <div>
            <Link to={`/food/edit/${id}`}><h2>{name}: {foodSize}</h2></Link>
            <p>{category}</p>
            <p>{amount}</p>
            <button onClick={()=>{dispatch(removeFood({id}))}}>Remove</button>
        </div>
    )
}

 

export default connect()(FoodListItem)