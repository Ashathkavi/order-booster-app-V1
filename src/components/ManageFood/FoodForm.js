import React from 'react'
import CategoryModal from '../ManageFood/CategoryModal'
import moment from 'moment'

export default class FoodForm extends React.Component{
    
    constructor(props){

        super(props)
        
        this.state={
            modalIsOpen:false,
            category: !!props.food? props.food.category  : (!!props.foodlargeAvailable ? props.foodlargeAvailable.category :''),
            fullAvailability:props.food ? props.food.largeAvailability : (!!props.foodlargeAvailable ? true: false),
            foodSize:props.food ? props.food.foodSize : (!!props.foodlargeAvailable ? props.foodlargeAvailable.foodSize : 'regular'),
            name:props.food ? props.food.name :  (!!props.foodlargeAvailable ? props.foodlargeAvailable.name :''),
            description:props.food ? props.food.description : (!!props.foodlargeAvailable ? props.foodlargeAvailable.description :  ''),
            amount: props.food ? props.food.amount.toString() : '',
            createdAt: props.food ? props.food.createdAt : moment().valueOf(),
            error:''
        }
        console.log(this.state.category)
    }
    
    

    onSetSize = (e) => {
        this.setState(() => ({
            foodSize:e.target.value
        }))              
    }

    onAvailabilityChange = (e) => {
        //console.log('e.target.value', 'true' === e.target.value)
        if(e.target.value==='false' && this.state.foodSize === 'full'){
            this.setState(() => ({
                foodSize:'regular'
            }))
        }
            
        this.setState(() => ({
            fullAvailability:(e.target.value === 'true')
        }))
    }

    handleCloseModal = () => {    
        this.setState(()=>({
            modalIsOpen:false
        }))
    }
    handleOpenModal = () => {       
        //console.log('open')
        this.setState(()=>({
            modalIsOpen:true
        }))
    }
    handleCategory = (category) => { 
        //console.log('category', category)
        //document.getElementById('selectedCat').value=category; 
        this.setState(()=>({
            category
        }))
    }
    onNameChange = (e) => {
        this.setState(()=>({
            name:e.target.value
        }))
    }
    onDescriptionChange = (e) => {
        this.setState(()=>({
            description:e.target.value
        }))
    }
    onAmountChange = (e) => {
        this.setState(()=>({
            amount:e.target.value
        }))
        //document.getElementById('textInput').value=e.target.value; 
    }

    onCheckFoodsPresents = (foodName, foodSize) => {
        let presents = false
        this.props.foods.map((food)=>{
            if(food.name.toLowerCase() === foodName.toLowerCase() && food.foodSize === foodSize){
                presents = true
                if(!!this.props.food && (this.props.food.name.toLowerCase() === foodName.toLowerCase() && this.props.food.foodSize === foodSize)){
                    presents = false
                }
            }
        })
        return presents
    }

    onSubmit = (e) => {
        e.preventDefault()
        if(!this.state.name || !this.state.category || !this.state.foodSize ){
            this.setState(()=>({
                error:'The field such as NAME, CATEGORY, FULL_AVAILABILITY and FOODSIZE are mandatoroy to fill'
            }))
        }else if(this.onCheckFoodsPresents(this.state.name, this.state.foodSize)){
            this.setState(()=>({
                error:'The Submitted Food is already Available'
            }))
        }else{
            this.setState(()=>({error:''}))
            this.props.onSubmit({
                name:this.state.name,
                amount:parseInt(this.state.amount),
                createdAt:this.state.createdAt,
                description:this.state.description,
                largeAvailability:this.state.fullAvailability,
                foodSize:this.state.foodSize,
                category:this.state.category
            })
        }
    }



    //this.handleSizeRadio()
    render(){
        console.log(this.props,'this.props')
        console.log(this.state,'this.state')
        // console.log('state.modalIsOpen',this.state.modalIsOpen)
        // console.log('state.category',this.state.category)
        // console.log('state.fullAvailability',this.state.fullAvailability)
        // console.log('state.foodSize',this.state.foodSize)
        // console.log('state.description',this.state.description)
        // console.log('state.name',this.state.name)
        return (
                
                
                <form onSubmit={this.onSubmit} className="form">
                    
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                    <div>
                        { !this.state.category && <span>  Please Select a Food Category First &nbsp;&nbsp;&nbsp;&nbsp;</span>}
                        <button disabled={!!this.props.foodlargeAvailable} type="button" onClick={this.handleOpenModal}> Select Category</button>
                        &nbsp;&nbsp;
                        <input disabled={!!this.props.foodlargeAvailable}  type='text' id='selectedCat' value={this.state.category} disabled required></input>
                    </div>
                    <input 
                        type="text" 
                        placeholder="Name" 
                        disabled={  !this.state.category || !!this.props.foodlargeAvailable}
                        value={this.state.name}
                        onChange={this.onNameChange}
                        className="text-input"
                    />
                    <div>
                        <input type='text' id='textInput' value={this.state.amount} disabled></input>
                        <input  type='range' min='0' max='2000' disabled={!this.state.category} step='10' value={this.state.amount} onChange={(e)=>{
                            this.onAmountChange(e)
                            //this.updateTextInput(e.target.value)
                        }}/>
                    </div>
                    


                    <div >Is Full postion available &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;

                        <input disabled={!!this.props.foodlargeAvailable} type="radio" value={true} checked={this.state.fullAvailability === true} onChange={this.onAvailabilityChange} /> Yes &nbsp;&nbsp;&nbsp;
                        <input disabled={!!this.props.foodlargeAvailable} type="radio" value={false} checked={this.state.fullAvailability === false} onChange={this.onAvailabilityChange} /> No
                    </div>
                    <div hidden={!this.state.fullAvailability} >Set the size of the potion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;

                        <input disabled={!!this.props.foodlargeAvailable} type="radio" value='regular' checked={this.state.foodSize === 'regular'} onChange={this.onSetSize} /> Normal &nbsp;&nbsp;&nbsp;
                        <input disabled={!!this.props.foodlargeAvailable} type="radio" value='full' checked={this.state.foodSize === 'full'} onChange={this.onSetSize} /> Full
                    </div>

                    
                                
                    <CategoryModal modalIsOpen={this.state.modalIsOpen} handleCloseModal={this.handleCloseModal} handleCategory={this.handleCategory}/>
                    
                    <textarea 
                        placeholder="Add a description about this food" 
                        disabled={!this.state.category}
                        value={this.state.description}
                        onChange={this.onDescriptionChange}   
                        className="textarea"
                    />
                    
                    <div>
                        <button className="button">Save Food</button>

                    </div>

                </form>
        )
    }
}