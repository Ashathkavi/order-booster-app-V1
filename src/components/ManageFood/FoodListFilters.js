import React, {useState} from 'react'
import {connect} from 'react-redux'
import {setBoundryAmount, setCatTextFilter, setNameFilter, setSize} from '../../actions/foodFilters'
import CategoryModal from '../ManageFood/CategoryModal' 

export const FoodListFilters = (props) => {
    // States to handle category MODAL 
    const [modalIsOpen, setModalIsOpen] = useState(false)
    const [boundryAmountDisplay, setboundryAmountDisplay] = useState('')
   
    const handleCloseModal = () => setModalIsOpen(false);
    const handleOpenModal = () => setModalIsOpen(true);

    const handleCategory = (category) => {
        props.setCatTextFilter(category)
    }

    const onNameChange = (e) => {
        e.persist()
        //console.log(props)
        props.setNameFilter(e.target.value)
    }

    const onSizeChange = (e) => {
        e.persist()
        props.setSize(e.target.value)
    }

    const onBoundryAmountChange = (e) => {
        e.persist()
        props.setBoundryAmount(e.target.value)
        setboundryAmountDisplay(e.target.value)
        //document.getElementById('textInput').value=e.target.value;
    }
    

    return(
        <div>
            {/*setNameFilter filter*/}
            <input type="text" value={props.filters.name} onChange={onNameChange}/>

            {/*setSize filter*/}
            <select value={props.filters.size} onChange={onSizeChange}>
                <option value=''>All</option>
                <option value="regular">Regular</option>
                <option value="full">Full</option>
            </select>

            {/*setCatTextFilter filter*/}
            <button onClick={handleOpenModal}>{props.filters.catText ? props.filters.catText : 'Select Category'}</button>           
            <CategoryModal modalIsOpen={modalIsOpen} handleCloseModal={handleCloseModal} handleCategory={handleCategory}/>

            {/*setBoundryAmount filter*/}
            <input type='range' min='0' max='2000' step='10' value={props.filters.boundryAmount} onChange={onBoundryAmountChange}/>
            <input type='text' id='textInput' disabled value={boundryAmountDisplay}></input>

        </div>
)}

const mapStateToProps = (state) => ({filters:state.foodFilters})

const mapDispatchToProps = (dispatch) => ({
    
    setBoundryAmount: (amount) => dispatch(setBoundryAmount(amount)),
    setSize: (size) => dispatch(setSize(size)),
    setCatTextFilter: (catText) => dispatch(setCatTextFilter(catText)),
    setNameFilter: (name) => dispatch(setNameFilter(name))

})

export default connect(mapStateToProps, mapDispatchToProps)(FoodListFilters)