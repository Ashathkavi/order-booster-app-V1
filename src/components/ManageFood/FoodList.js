import React from 'react'
import {connect} from 'react-redux'
import FoodListItem from '../ManageFood/FoodListItem'
import selectFood from '../../selectors/foods'

export const FoodList = (props) => {
    //console.log(props)
    return (
    <div>
        {
            props.foods.length === 0 ? (
                <p>No Foods to Show</p>
            ):(
                props.foods.map((food)=>{
                    return <FoodListItem key={food.id} {...food}/>
                })
            )
        }
    </div>
)}

const mapStateToProps = (state) => {
    return {
        foods:selectFood(state.foods, state.foodFilters),
    }
}

export default connect(mapStateToProps)(FoodList)