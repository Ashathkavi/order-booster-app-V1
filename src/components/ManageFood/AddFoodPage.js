import React from 'react'
import FoodForm from '../ManageFood/FoodForm'
import {connect} from 'react-redux'
import {addFood} from '../../actions/foods'

export const AddFoodPage = (props) => {
    //console.log(props.history)
    const onSubmit = (food) => {
        props.addFood(food)
        props.history.push('/food')
    }
    return(
        <div>
            <h1>Add Food</h1>
            <FoodForm onSubmit={onSubmit}/>
        </div>
)}

const mapDispatchToProps = (dispatch) => ({
    addFood: (food) => dispatch(addFood(food))
})

export default connect(undefined, mapDispatchToProps)(AddFoodPage)