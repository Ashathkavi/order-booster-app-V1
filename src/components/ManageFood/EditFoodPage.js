import React from 'react'
import {connect} from 'react-redux'
import { editFood, startRemoveFood} from '../../actions/foods'
import FoodForm from '../ManageFood/FoodForm'

export const EditFoodPage = (props) => {
    //console.log(props)
    const onSubmit = (food) => {
        props.editFood(props.food.id ,food)
        props.history.push('/food')
    }

    const onRemove = () => {
        props.startRemoveFood(props.food.id)
        props.history.push('/food')
    }
    return(
        <div>
           <FoodForm food={props.food} onSubmit={onSubmit} />
           <button onClick = {onRemove}>Remove</button>
        </div>
)}

const mapDispatchToProps = (dispatch) => ({
    editFood: (id, food) => dispatch(editFood(id, food)),
    startRemoveFood: (id) => dispatch(startRemoveFood({id}))
})

const mapStateToProps = (state, props) => {
    console.log(state.foods)
    return {
        
        food:state.foods.find((food)=> food.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditFoodPage)