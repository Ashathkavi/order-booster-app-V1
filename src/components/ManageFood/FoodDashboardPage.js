import React from 'react'
import {Link} from 'react-router-dom'
import FoodList from '../ManageFood/FoodList'
import FoodListFilters from './FoodListFilters'

const FoodDashboardPage = () => (
    <div>
        <FoodListFilters/>
        <FoodList/>
    </div>
)

export default FoodDashboardPage