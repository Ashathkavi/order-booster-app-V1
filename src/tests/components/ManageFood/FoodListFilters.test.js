import React from 'react'
import {shallow} from 'enzyme'
import {FoodListFilters} from '../../../components/ManageFood/FoodListFilters'
import {foodFilters, altFoodFilters} from '../../../fixtures/sampleFoodFilters'
import CategoryModal from '../../../components/ManageFood/CategoryModal'
import Modal from 'react-modal'

let setNameFilterSpy, setCatTextFilterSpy, setBoundryAmountSpy, setSizeSpy, wrapper

beforeEach(() => {
    setNameFilterSpy = jest.fn()
    setCatTextFilterSpy = jest.fn()
    setBoundryAmountSpy = jest.fn()
    setSizeSpy = jest.fn()
    wrapper = shallow(
        <FoodListFilters 
            filters={foodFilters}
            setBoundryAmount = {setBoundryAmountSpy}
            setCatTextFilter = {setCatTextFilterSpy}
            setNameFilter ={setNameFilterSpy}
            setSize = {setSizeSpy}
        />
    )
})

test('should render FoodList Coorectly', ()=>{
    expect(wrapper).toMatchSnapshot()
})

test('should render FoodList with alternate food filter', ()=>{
    wrapper.setProps({
        filters:altFoodFilters
    })
    expect(wrapper).toMatchSnapshot()
})


test('should handle onBoundryChange', () => {
    const value = 500
    wrapper.find('input').at(1).simulate('change', {
        
        persist: () => { },
        target:{value}
    })
    expect(setBoundryAmountSpy).toHaveBeenLastCalledWith(value)
})

test('should handle setNameFilter', () => {
    const value = 'soup'
    wrapper.find('input').at(0).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(setNameFilterSpy).toHaveBeenLastCalledWith(value)
})


test('should handle setSize', () => {
    const value = 'full'
    wrapper.find('select').first().simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(setSizeSpy).toHaveBeenLastCalledWith(value)
})


test('should handle category modal opening', () => {
    wrapper.find('button').first().simulate('click')    
    expect(wrapper.find(CategoryModal).prop('modalIsOpen')).toBe(true)
})


test('should handle category modal clossing', () => {
    wrapper.find('button').first().simulate('click')
    wrapper.find(CategoryModal).prop('handleCloseModal')();    
    expect(wrapper.find(CategoryModal).prop('modalIsOpen')).toBe(false)
})

test('should handle onchange category', () => {
    const value = 'fried rice'
    wrapper.find(CategoryModal).prop('handleCategory')(value);    
    expect(setCatTextFilterSpy).toHaveBeenLastCalledWith(value)
})

