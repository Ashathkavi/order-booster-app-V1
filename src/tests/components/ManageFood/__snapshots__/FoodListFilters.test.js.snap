// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FoodList Coorectly 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="regular"
    >
      Regular
    </option>
    <option
      value="full"
    >
      Full
    </option>
  </select>
  <button
    onClick={[Function]}
  >
    Select Category
  </button>
  <CategoryModal
    handleCategory={[Function]}
    handleCloseModal={[Function]}
    modalIsOpen={false}
  />
  <input
    max="2000"
    min="0"
    onChange={[Function]}
    step="10"
    type="range"
    value={2000}
  />
  <input
    disabled={true}
    id="textInput"
    type="text"
    value=""
  />
</div>
`;

exports[`should render FoodList with alternate food filter 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value="ch"
  />
  <select
    onChange={[Function]}
    value="reg"
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="regular"
    >
      Regular
    </option>
    <option
      value="full"
    >
      Full
    </option>
  </select>
  <button
    onClick={[Function]}
  >
    fri
  </button>
  <CategoryModal
    handleCategory={[Function]}
    handleCloseModal={[Function]}
    modalIsOpen={false}
  />
  <input
    max="2000"
    min="0"
    onChange={[Function]}
    step="10"
    type="range"
    value={2000}
  />
  <input
    disabled={true}
    id="textInput"
    type="text"
    value=""
  />
</div>
`;
